client/blb/client.go

```diff
@@ -1113,18 +1129,117 @@ func (cli *Client) readOneTractReplicated(
 
 	var badVersionHost string
 
+	reqID := core.GenRequestID()
 	order := rand.Perm(len(tract.Hosts))
+	orderCh := make(chan int, len(tract.Hosts))
+
+	// This goroutine hands out the order in which to process reads.
+	// This ensures that when goroutines are done processing the initial backup
+	// phase, we can know which hosts still need to be processed sequentially.
+	go func() {
+		for _, n := range order {
+			orderCh <- n
+		}
+		close(orderCh)
+	}()
 
 	err := core.ErrAllocHost // default error if none present
-	for _, n := range order {
+	if cli.backupReadState.BackupReadBehavior.Enabled {
+		ch := make(chan tractResult)
+
+		ctx, cancel := context.WithCancel(ctx)
+		defer cancel()
+
+		maxNumBackups := cli.backupReadState.BackupReadBehavior.MaxNumBackups
+		delay := cli.backupReadState.BackupReadBehavior.Delay
+
+		// report bad tractservers
+		var badTSReporter []errorResult
+
+		readFunc := func(orderCh chan int) {
+			n := <-orderCh
+			host := tract.Hosts[n]
+			if host == "" {
+				log.V(1).Infof("read %s from tsid %d: no host", tract.Tract, tract.TSIDs[n])
+				ch <- tractResult{0, 0, err, badVersionHost}
+				return
+			}
+
+			// TODO(eric): fill in otherHosts when ts-ts cancellation is done.
+			read, err := cli.tractservers.ReadInto(ctx, host, nil, reqID, tract.Tract, tract.Version, thisB, thisOffset)
+			if err == core.ErrVersionMismatch {
+				badVersionHost = host
+			}
+			if err != core.NoError && err != core.ErrEOF {
+				badTSReporter = append(badTSReporter, errorResult{tract.Tract, host, err})
```

Comment:
This will probably be a race, can use a channel or wrap in a lockable struct, if we want to go this way.